test/control-plane/services/lbmap_validation.go

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package services
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/netip"
+	"os"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/pmezard/go-difflib/difflib"
+	"golang.org/x/exp/constraints"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+
+	fakeDatapath "github.com/cilium/cilium/pkg/datapath/fake"
+	lb "github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/testutils/mockmaps"
+	controlplane "github.com/cilium/cilium/test/control-plane"
+)
+
+func NewGoldenServicesTest(t *testing.T, nodeName string) *controlplane.ControlPlaneTestCase {
+	return controlplane.NewGoldenTest(t, nodeName, newGoldenLBMapValidator)
+}
+
+type goldenLBMapValidator struct {
+	expectedFile string
+	update       bool
+}
+
+func newGoldenLBMapValidator(stateFile string, update bool) controlplane.Validator {
+	var v goldenLBMapValidator
+	var stepNum int
+	fmt.Sscanf(path.Base(stateFile), "state%d.yaml", &stepNum)
```

Comment:
should we check for errors from Sscanf, might cause some confusing errors down the line?

Reply:
Added the check.